src/cic/repl.clj

```diff
@@ -1,86 +1,89 @@
 (ns cic.repl
-  (:require [clojure.string :as str]
-            [clojure.set :as cs]
-            [clojure.test.check.random :as r]
-            [cic.core :as core]
-            [clojure.data.csv :as data-csv]
-            [clojure.java.io :as io]
+  (:require [cic.episodes :as episodes]
+            [cic.io.read :as read]
+            [cic.io.write :as write]
             [cic.model :as model]
+            [cic.periods :as periods]
             [cic.projection :as projection]
-            [clj-time.format :as f]))
+            [cic.random :as rand]
+            [cic.summary :as summary]
+            [cic.time :as time]
+            [cic.validate :as validate]
+            [clojure.set :as cs]))
 
-(def date-format
-  (f/formatter :date))
-
-(defn date->str
-  [date]
-  (f/unparse date-format date))
-
-(defn write-projection-tsv
-  [outfile projection]
-  (let [fields (juxt (comp date->str :date)
-                     :cost
-                     (comp :actual)
-                     (comp :lower :total)
-                     (comp :q1 :total)
-                     (comp :median :total)
-                     (comp :q3 :total)
-                     (comp :upper :total)
-                     :Q2 :K2 :Q1 :R2 :P2 :H5 :R5 :R1 :A6 :P1 :Z1 :S1 :K1 :A4 :T4 :M3 :A5 :A3 :R3 :M2 :T0
-                     #(get % 0) #(get % 1) #(get % 2) #(get % 3) #(get % 4) #(get % 5) #(get % 6) #(get % 7) #(get % 8) #(get % 9)
-                     #(get % 10) #(get % 11) #(get % 12) #(get % 13) #(get % 14) #(get % 15) #(get % 16) #(get % 17) #(get % 18))
-        placements [:Q2 :K2 :Q1 :R2 :P2 :H5 :R5 :R1 :A6 :P1 :Z1 :S1 :K1 :A4 :T4 :M3 :A5 :A3 :R3 :M2 :T0]
-        ages (range 0 19)
-        headers (concat ["Date" "Cost" "Actual" "Lower CI" "Lower Quartile" "Median" "Upper Quartile" "Upper CI"]
-                        (map name placements)
-                        (map str ages))]
-    (with-open [writer (io/writer outfile)]
-      (data-csv/write-csv writer (concat [headers] (map fields projection))))))
-
-(defn project
-  [episodes project-from project-to joiners-model duration-model seed]
-  (let [periods (core/episodes->periods episodes)]
-    (projection/projection periods
-                           project-from project-to
-                           joiners-model duration-model
-                           seed
-                           10)))
+(defn load-model-inputs
+  "A useful REPL function to load the data files and convert them to  model inputs"
+  []
+  (hash-map :periods (-> (read/episodes "data/episodes.csv")
```

Comment:
So if this is reading in episodes.csv perhaps we should translate the Python code that currently preps this file from the SSDA903 data, @otfrom ?

Reply:
@seb231 yes we should. I'd like to reduce the number of tools we need to use to get to where we want to be.

Reply:
Discussed elsewhere. Need to translate R and Python code to single independent Clojure tool for prepping SSDA903